ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,404 @@
+        let view = DimmingView(type: .black)
+        view.isUserInteractionEnabled = false
+        return view
+    }()
+
+    private lazy var panGestureRecognizer: UIPanGestureRecognizer = {
+        let recognizer = UIPanGestureRecognizer()
+        recognizer.addTarget(self, action: #selector(handlePan))
+        return recognizer
+    }()
+
+    private lazy var resizingHandleView: ResizingHandleView = {
+        let resizingHandleView = ResizingHandleView()
+        resizingHandleView.isAccessibilityElement = true
+        resizingHandleView.accessibilityTraits = .button
+        resizingHandleView.isUserInteractionEnabled = true
+        resizingHandleView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleResizingHandleViewTap)))
+        return resizingHandleView
+    }()
+
+    @objc public init(with contentViewController: UIViewController) {
+        self.contentViewController = contentViewController
+        super.init(nibName: nil, bundle: nil)
+        addChild(contentViewController)
+        contentViewController.didMove(toParent: self)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+// MARK: - Lifecycle methods
+
+    open override func viewDidDisappear(_ animated: Bool) {
+        removeDimmingView()
+        super.viewDidDisappear(animated)
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = Colors.NavigationBar.background
+
+        view.layer.shadowColor = Constants.Shadow.color
+        view.layer.shadowOpacity = Constants.Shadow.opacity
+        view.layer.shadowRadius = Constants.Shadow.radius
+        view.layer.shadowOffset = Constants.Shadow.offset
+        view.layer.cornerRadius = Constants.cornerRadius
+        view.layer.cornerCurve = .continuous
+        updateMaskedCorners()
+
+        view.addGestureRecognizer(panGestureRecognizer)
+        panGestureRecognizer.delegate = self
+
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addArrangedSubview(resizingHandleView)
+        containerView.addArrangedSubview(contentViewController.view)
+        view.addSubview(containerView)
+        NSLayoutConstraint.activate([
+            containerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            containerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            containerView.topAnchor.constraint(equalTo: view.topAnchor),
+            containerView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+
+        updateResizingHandleViewAccessibility()
+    }
+
+    public override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        if animator?.state == .active {
+            // panning started, don't update the height
+            return
+        }
+        updateFrame()
```

Comment:
This seems a bit scary - it looks like this could call setNeedsLayout internally to updateFrame and calling that from viewWillLayoutSubviews is only needed when we require a 2-pass layout where we can carefully and predictably avoid a loop. Is that actually the case here? If so we should probably add comments to help preserve that potential source of fragility